From 83f8703511052adda7c9dd816885e18795a52afd Mon Sep 17 00:00:00 2001
From: Robert Richter <rrichter@cavium.com>
Date: Fri, 2 Jan 2015 13:37:57 +0100
Subject: [PATCH 078/184] AHCI: Use gotos instead of while

Source: Cavium Networks, Inc.
MR: 13384
Type: Integration
Disposition: Merged from THUNDERX Tree
ChangeID: 88e60971aebe549a6709694c750ff1a1dd96fb94
Description:

Signed-off-by: Robert Richter <rrichter@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from octeon-linux-kernel-patches-SDK-3.1.2-release]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/ata/libahci.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/ata/libahci.c b/drivers/ata/libahci.c
index 9ca07c3..dc8c016 100644
--- a/drivers/ata/libahci.c
+++ b/drivers/ata/libahci.c
@@ -1867,11 +1867,9 @@ static irqreturn_t ahci_single_irq_intr(int irq, void *dev_instance)
 
 	/* sigh.  0xffffffff is a valid return from h/w */
 	irq_stat = readl(mmio + HOST_IRQ_STAT);
-
-	do {
-
+redo:
 	if (!irq_stat)
-		return IRQ_NONE;
+		goto out;
 
 	irq_masked = irq_stat & hpriv->port_map;
 
@@ -1911,8 +1909,8 @@ static irqreturn_t ahci_single_irq_intr(int irq, void *dev_instance)
 
 	spin_unlock(&host->lock);
 
-	} while(irq_stat);
-
+	goto redo;
+out:
 	VPRINTK("EXIT\n");
 
 	return IRQ_RETVAL(handled);
-- 
1.9.1

